version: "3.8"
services:
    frontend:
        restart: on-failure
        build:
            dockerfile: ./docker/Dockerfile.dev
            context: ./frontend
        volumes:
            - ./frontend/package.json:/frontend/package.json
            - ./frontend/src:/frontend/src/
            - ./frontend/public:/frontend/public/
        stdin_open: true
        environment:
            - SCETA_API_PATH=api/predictions
    server:
        restart: always
        build:
            dockerfile: ./docker/Dockerfile
            context: ./server
        command:
            - --config=/server/data.yaml
            - --host=0.0.0.0
            - --port=8001
        volumes:
            - ./server/data.yaml:/server/data.yaml
            - ./server/app.py:/server/app.py
            - ./server/modules:/server/modules
        environment:
            - WATCHFILES_FORCE_POLLING=true
    nginx:
        image: nginx:alpine
        volumes:
            - ./nginx.conf:/etc/nginx/nginx.conf
networks: 
  default:
    external: 
      name: sce
